### PR TITLE
Update wgpu and WebGPU CTS to include copy validation tests

### PR DESCRIPTION
**Pull request by kunalmohan**
r?@kvark

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] There are tests for these changes

### CODE REVIEW

**Comment by kunalmohan** [comment]
@bors-servo try=wpt-mac

Let's see if there are any unexpected crashes on macOS in the new tests.

**Comment by kunalmohan** [comment]
@kvark Only the first commit is worth reviewing, rest is just copy-paste of new tests.

File: components/script/dom/gpudevice.rs [hunk 1]
```diff
@@ -739,8 +764,12 @@ impl GPUDeviceMethods for GPUDevice {
     fn CreateTexture(&self, descriptor: &GPUTextureDescriptor) -> DomRoot<GPUTexture> {
         let size = convert_texture_size_to_dict(&descriptor.size);
         let desc =
-            wgt::TextureUsage::from_bits(descriptor.usage).map(|usg| wgt::TextureDescriptor {
-                label: descriptor.parent.label.as_ref().map(|s| s.to_string()),
+            wgt::TextureUsage::from_bits(descriptor.usage).map(|usg| wgpu_res::TextureDescriptor {
+                label: descriptor
```
Comment (kvark) [inline components/script/dom/gpudevice.rs:768]: it looks like you could benefit from a small helper function to get that label

File: components/script/dom/gputexture.rs [hunk 2]
```diff
@@ -156,14 +141,16 @@ impl GPUTextureMethods for GPUTexture {
...
+            Some(resource::TextureViewDescriptor {
                 label: descriptor
                     .parent
                     .label
                     .as_ref()
-                    .map(|s| String::from(s.as_ref())),
-                format: convert_texture_format(format),
-                dimension: convert_texture_view_dimension(dimension),
+                    .map(|l| Cow::Owned(l.to_string())),
+                format: descriptor.format.map(|fr| convert_texture_format(fr)),
```
Comment (kvark) [inline components/script/dom/gputexture.rs:150]: nit: can just do `map(convert_texture_format)`

File: components/script/dom/gputexture.rs [hunk 3]
```diff
@@ -156,14 +141,16 @@ impl GPUTextureMethods for GPUTexture {
...
+                dimension: descriptor
+                    .dimension
+                    .map(|dm| convert_texture_view_dimension(dm)),
```
Comment (kvark) [inline components/script/dom/gputexture.rs:153]: and here

**Review by kvark — APPROVED** [review]

**Comment by kvark** [comment]
notes can be addressed in a follow-up :)
@bors-servo r+

**Pull request merged**